src/app/organizations/state/collections.service.ts

```diff
@@ -101,8 +101,11 @@ export class CollectionsService {
           this.collectionsStore.setActive(collection.id);
           this.organizationService.updateOrganizationFromID(collection.organizationID);
         },
-        () => {
+        (collection: Collection) => {
```

Comment:
I think you can just have `(error: HttpErrorResponse)`, then you don't need do the casting on line 106. https://angular.io/guide/http#error-handling. It's weird, I'd never looked at it before, but the IDE/compiler doesn't seem to be able to infer that.

That said, the doc I linked to says it returns an error object, so I think it's safe to declare it as an ErrorResponseObject. It's definitely not going to be a `Collection` though, so I wouldn't declare it as that.

Reply:
Ah that makes a lot of sense, thank you